### DISCUSSION TITLE
Don't Understand how do the tests

### DISCUSSION THREAD

**Discussion by arbdev95** [Cyfrin/foundry-full-course-cu#111] [Q&A] [answered]
Hello guys,

i'm at lesson 4, but i'm trying to do the tests from lesson 1, but i don't know what to do, because in the test of lesson 1, when i compile 1-Lesson.sol, it says "Error: not found AFoundryCourseChallenge.sol", what to do, or how it works in these tests?

**Comment by alymurtazamemon** [comment]
@arbdev95 Which test are you talking about? Lesson 1 does not have any tests!

And this `AFoundryCourseChallenge.sol` contract belongs to challenges contracts.

**Comment by arbdev95** [comment]
> @arbdev95 Which test are you talking about? Lesson 1 does not have any tests!
> 
> And this `AFoundryCourseChallenge.sol` contract belongs to challenges contracts.

i mean the tests at the end of every lesson(the QRcode), or am i wrong?

**Reply by cromewar** [reply]
Could you be more specific please, finding hard what you mean.

**Reply by alymurtazamemon** [reply]
@arbdev95 Ah! you mean challenges, you do not need to copy them locally just call the `solveChallenge` function on sepolia etherscan.

**Comment by ShayanShamsi** [comment, marked as answer]
Hey, @arbdev95. I am assuming you are doing the challenges at the end of lesson 1 and you accidentally said lesson 4 in your question.
I have not attempted these challenges myself (yet) but I think you are misinterpreting how you are required to attempt the challenge. From what I could gather, you are **not supposed to compile 1-Lesson.sol**. The contract has already been deployed on the chain. For the challenge, you just need to call the `solveChallenge` function on the already deployed contract with your Twitter handle as the argument.

Here is how you might do that:
1. If you do not have a Twitter account, sign-up [here](https://twitter.com/i/flow/signup).
2. Go to the link you obtained by scanning the QR code for the challenge at the end of lesson 1. (I think [this one](https://sepolia.etherscan.io/address/0x25056312685339b49e1d1C5a0b72Ff9eff13AF77#code)).
3. Click on "Write Contract" which should take you [here](https://sepolia.etherscan.io/address/0x25056312685339b49e1d1C5a0b72Ff9eff13AF77#writeContract).
4. Click on "solveChallenge".
5. Enter your Twitter handle in the text field.
6. Press the "Write" button.
7. Complete the transaction.

**Please note that your wallet will need money to complete the transaction. If you are completing the challenge on the Sepolia test chain, then you can obtain Sepolia ETH by going to faucets.chain.link.**

As for the error that you were obtaining while compiling 1-Lesson.sol, it might be happening because you were trying to compile 1-Lesson.sol alone. If you look at the source code for 1-Lesson.sol, you can see that it is importing `AFoundryCourseChallenge.sol`, and that import would fail if you do not have `AFoundryCourseChallenge.sol` in the same directory. You probably did not have `AFoundryCourseChallenge.sol` in your directory which caused the compilation to fail.

I hope this helps and let me know if you have any more questions!

**Reply by arbdev95** [reply]
Hey @ShayanShamsi  everything clear, thank you.

**Reply by BLTC-520** [reply]
This freaking helps a lot. Thanks!

**Reply by GuilleMualem** [reply]
Hello guys, I have followed the thread of this conversation but in my case something else is failing and I don't understand what it is.
When I click on write and call the function as you say, the following signs appear
![screen001](https://github.com/Cyfrin/foundry-full-course-f23/assets/157833172/b4fdd0f2-05bc-41e9-a36a-cdba66d11146)

Then I click "continue anyway"... and I get this message on etherscan.
![screenlesson1](https://github.com/Cyfrin/foundry-full-course-f23/assets/157833172/9c0259f7-ccf4-43ac-bdbe-fb2837c83dcf)
 
I don't understand if it is saying that I don't have enough gas in Sepolia, my Sepolia amount at the moment is 0.4, I understand that it is more than enough to make the call.
If anyone can suggest me what to do 🙏🏼. Thank you!

**Reply by EngrPips** [reply]
Hello @GuilleMualem  did you pass in the required argument to the function call ?

**Reply by EngrPips** [reply]
@GuilleMualem , so i went ahead to do the challenge with my wallet passing in my twitter handle as an argument to the `solveChallenge` function call and it was pretty successful, below are screenshot of my transactions 

**ME APPROVING THE TRANSACTION**
<img width="421" alt="Screenshot 2024-03-04 at 5 31 13 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/83e61dd5-3bcb-4cea-a416-4540f2f7483f">

**THE TRANSACTION DETAILS**
<img width="1376" alt="Screenshot 2024-03-04 at 5 32 04 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/f172f54c-1ab0-4a32-9b56-9d778ddc4026">

So @GuilleMualem  i would suggest you make sure that you have enough sepolia ether in the wallet you are using to call the `solveChallenge` function, i would say at least 1 sepolia ETH because it seem that gas fee on sepolia are on spike lately. Also make sure you pass in your twitter handle as an argument to the function call.

**Comment by GuilleMualem** [comment]
@EngrPips 
Thanks for your answer! What does it mean if I passed the required argument to the function call? I did what is indicated step by step in the comments above @ShayanShamsi. I have claimed more faucet from Sepolia, now I have 0.99 .. I must wait until tomorrow to have 0.5 more. If anyone wants to send me something to complete the 1 Sepolia eth, this is my public address:
0x87221e4095F09803fA07217Baf8572d9F0fFc282
Oh and just so you know, I already tried again with the current amount of Sepolia I have and I'm still getting the same error.
Thanks for the support @EngrPips !

**Reply by GuilleMualem** [reply]
@EngrPips I have tried having more Sepolia (1.24..) and the same error continues. Maybe I need a minimum of real money on ethereum mainnet?

**Reply by EngrPips** [reply]
Hello @GuilleMualem, did you pass in your twitter handle as an argument when calling the `solveChallenge` function ?
You don't need any real money to solve the challenge, I didn't have any real money in my account when i did the challenge and solved it.

Maybe make a step by step explanation and screenshot of what you are doing till you get to the point where you get the error and also takes a screenshot of the error as well , like that it would more easier for anybody to help and figure out what you are doing wrong.

Also are you sure you are interacting with the correct contract ? , This is the contract address `0x25056312685339b49e1d1C5a0b72Ff9eff13AF77` ,  You can search it on the `sepolia etherscan` and go to the write section of the contract to interact with the `solveChallenge` function.

I would be waiting to hear back from you.

**Reply by GuilleMualem** [reply]
Hi @EngrPips . I've been a little too busy to answer but here I am. Thanks for your response again. 

I leave you step by step of what I have been doing with screenshots.

![1](https://github.com/Cyfrin/foundry-full-course-f23/assets/157833172/feb8a17f-483b-438c-8f58-7dfd3be4b0e3)
First capture, without connecting to the wallet, so you can see that I am in the contract and in the right place.
![2](https://github.com/Cyfrin/foundry-full-course-f23/assets/157833172/8c7a3bf0-5d6b-4e00-b97b-4cf48625824b)
Connected wallet. You can see that there is enough Sepolia.
![3](https://github.com/Cyfrin/foundry-full-course-f23/assets/157833172/d31ad661-b23b-4743-9de4-c817b3e625d1)
I write my twitterhandle before clicking write.
![4](https://github.com/Cyfrin/foundry-full-course-f23/assets/157833172/118c3ba9-62bf-4d60-b3b5-54f429d61c69)
Then this sign appears, and I click "I want to continue anyway."
![5](https://github.com/Cyfrin/foundry-full-course-f23/assets/157833172/8d100c2d-5b74-427a-a914-23a26af9e8f8)
And then there is the last step with the error.
I also tried changing the gas rates and setting them to low, even high. And nothing has worked.
![6](https://github.com/Cyfrin/foundry-full-course-f23/assets/157833172/c4969de8-cb09-4d64-b29c-55a7b504812d)


Thanks for trying to help!

**Reply by EngrPips** [reply]
Hello @GuilleMualem , when you tried to enter your twitter handle please input `"@GuilleMualem"` instead of `"GuilleMualem"` and lets see if it would work. also wrap your twitter handle in a string `""`

**Reply by EngrPips** [reply]
Hello @GuilleMualem, i think i know what the problem is. So if you had minted to particular address you wouldn't be able to mint to that address anymore. maybe a transaction had already be successful for the address you are using to solve the challenge which you didn't realize. however to ensure you are able to mint since this your current address is reverting, i would say set up a new address which you have never used to call the nft contract and try solving the challenge with that new address, i believe that would work. Let me know when you try.
